packages/orbit-components/src/TextLink/index.tsx

```diff
@@ -78,16 +78,23 @@ const getSizeToken = ({ theme, size }: { theme: Theme; size: Common.Size }): str
   return sizeTokens[size];
 };
 
+const getIconSize = (theme: Theme, size: Props["size"]) => {
+  if (size === SIZE_OPTIONS.LARGE)
+    return { width: theme.orbit.widthIconLarge, height: theme.orbit.heightIconLarge };
+  if (size === SIZE_OPTIONS.SMALL)
+    return { width: theme.orbit.widthIconSmall, height: theme.orbit.heightIconSmall };
+  return { width: theme.orbit.widthIconMedium, height: theme.orbit.heightIconMedium };
+};
+
 const StyledIconContainer = styled(({ children, className }) => (
   <span className={className}>{children}</span>
 ))`
-  ${({ theme }: { theme: Theme }) => css`
+  ${({ theme, size }: { theme: Theme; size: Props["size"] }) => css`
     display: flex;
     align-items: center;
 
     svg {
-      width: ${theme.orbit.widthIconSmall};
-      height: ${theme.orbit.heightIconSmall};
+      ${getIconSize(theme, size)};
```

Comment:
```suggestion
      ${getIconSize};
```

Reply:
I like the previous suggestion but on this one I'd rather do `getIconSize({theme, size})`, like we do on `getSizeToken`